cmd/appengine/appengine.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2019 Ispirata Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appengine
+
+import (
+	"errors"
+
+	"github.com/astarte-platform/astartectl/client"
+
+	"github.com/astarte-platform/astartectl/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// AppEngineCmd represents the appEngine command
+var AppEngineCmd = &cobra.Command{
+	Use:               "appengine",
+	Short:             "Interact with AppEngine API",
+	Long:              `Interact with AppEngine API.`,
+	PersistentPreRunE: appEnginePersistentPreRunE,
+}
+
+var realm string
+var appEngineJwt string
+var realmManagementJwt string
+var astarteAPIClient *client.Client
+
+func init() {
+	AppEngineCmd.PersistentFlags().StringP("realm-key", "k", "",
+		"Path to realm private key used to generate JWT for authentication")
+	AppEngineCmd.MarkPersistentFlagFilename("realm-key")
+	AppEngineCmd.PersistentFlags().String("appengine-url", "",
+		"AppEngine API base URL. Defaults to <astarte-url>/appengine.")
+	viper.BindPFlag("appengine.url", AppEngineCmd.PersistentFlags().Lookup("appengine-url"))
+	AppEngineCmd.PersistentFlags().String("realm-management-url", "",
+		"Realm Management API base URL. Defaults to <astarte-url>/realmmanagement.")
+	viper.BindPFlag("realm-management.url", AppEngineCmd.PersistentFlags().Lookup("realm-management-url"))
```

Comment:
Since this is now used both here and in `realm.go`, I think the binding has to be moved in the respective `PersistentPreRunE` functions (like `realm.key` and `realm.name`), otherwise I think this won't work in one of the two commands